### ISSUE TITLE
Neovim 0.5 `:setf coq` required before syntax highlighting

### ISSUE THREAD

**Issue by Iron-E** [whonore/Coqtail#162]
Thanks for this plugin! I'm working on teaching PLT to beginners so I'm trying out Coq before actually forcing it on them.

I'm running into an issue with Neovim 0.5— namely, there's no highlighting. Here's a screenshot on 0.4.4:

![0.4.4](https://user-images.githubusercontent.com/36409591/101235765-432f4680-3699-11eb-8423-116910ffb697.png)

Here is 0.5:

![0.5](https://user-images.githubusercontent.com/36409591/101235773-5c37f780-3699-11eb-9701-c9dbfd18dd22.png)

Upon further inspection it appears that none of the tokens are being assigned to any syntax groups on 0.5.

**Comment by whonore** [comment]
Thanks for reporting this. Could you tell me how you installed Neovim and paste the output of `nvim --version`? I just tried downloading `nvim-linux64.tar.gz` from https://github.com/neovim/neovim/releases/tag/nightly and I can still see the syntax highlighting. I did notice that 0.5 is using a new [syntax engine](https://neovim.io/roadmap/) so maybe that's related?

**Comment by Iron-E** [comment]
I left instructions on how I install over at [this](https://github.com/nvim-lua/completion-nvim/issues/274#issuecomment-729961240) issue. The nightly is several weeks old at this point, and breaking changes have been made to 0.5 since its release. _That said_, I tried the appimage and it didn't work either so you don't have to bother with the above instructions.

I created a minimal rc file to test it:

```vim
" ## Begin Plugin Loading; change the path if you need to
call plug#begin(stdpath('cache').'/vim-plug')

Plug 'https://github.com/liuchengxu/space-vim-dark.git'
Plug 'https://github.com/whonore/Coqtail.git'

" ## Terminate plugin loading
call plug#end()
```

And ran `nvim -u minrc.vim -c 'colorscheme space-vim-dark' foo.coq`/`foo.v`

This yielded the same result. I notice the [`ftdetect`](https://github.com/whonore/Coqtail/blob/master/ftdetect/coq.vim) file specifies a `*.v`, but my Neovim picks this up as Verilog, not Coq. So, I created my own `ftdetect` in my config:

```vim
autocmd BufRead,BufNewFile *.coq set filetype=coq
```

However, this doesn't work either. I discovered I have to `:setf coq` _again_ in order for highlighting to appear:

![Highlighting!](https://user-images.githubusercontent.com/36409591/101260147-b1b9e600-3725-11eb-9daf-69def9029fee.png)

I also see that the first `|` gets picked up as `coqVernacPunctuation`, and the rest are picked up as `coqIndConstructor`. Is that intended?

**Comment by whonore** [comment]
I'm still not able to reproduce this issue using the Neovim 0.5 appimage and your minrc.vim. Maybe the output of `:scriptnames` could be useful to see if the Verilog syntax is overriding Coq, or if Coq isn't getting sourced at all. I attached the output that I see at the end of this comment.

As for the syntax questions, the syntax file is a pretty awful mess that was originally copied from a [mostly functional, but very outdated version](http://www.vim.org/scripts/script.php?script_id=2063) and has since had a bunch of ad-hoc fixes and additions taped onto it. I would still like to do #87 and basically start from scratch, but it's currently too big an undertaking for me. In the meantime, if you're feeling brave enough, I'd happily accept pull requests to fix some of these more isolated issues you pointed out. Or even just creating issues for them would help me keep track of them for when I do get around to working on it.

```
  1: ~/tmp/minrc.vim
  2: ~/.dotfiles/vim/autoload/plug.vim
  3: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/filetype.vim
  4: ~/.dotfiles/vim/bundle/Coqtail/ftdetect/coq.vim
  5: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/ftplugin.vim
  6: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/indent.vim
  7: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/syntax/syntax.vim
  8: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/syntax/synload.vim
  9: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/syntax/syncolor.vim
 10: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/gzip.vim
 11: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/health.vim
 12: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/man.vim
 13: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/matchit.vim
 14: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/pack/dist/opt/matchit/plugin/matchit.vim
 15: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/matchparen.vim
 16: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/netrwPlugin.vim
 17: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/rplugin.vim
 18: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/shada.vim
 19: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/spellfile.vim
 20: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/tarPlugin.vim
 21: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/tohtml.vim
 22: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/tutor.vim
 23: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/plugin/zipPlugin.vim
 24: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/ftplugin/verilog.vim
 25: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/indent/verilog.vim
 26: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/syntax/verilog.vim
 27: ~/.dotfiles/vim/bundle/Coqtail/ftplugin/coq.vim
 28: ~/.dotfiles/vim/bundle/Coqtail/autoload/coqtail.vim
 29: ~/.dotfiles/vim/bundle/Coqtail/autoload/coqtail/compat.vim
 30: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/autoload/provider/python3.vim
 31: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/autoload/provider/pythonx.vim
 32: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/autoload/remote/host.vim
 33: /tmp/.mount_nvim.apgB80f/usr/share/nvim/runtime/autoload/provider/python.vim
 34: ~/.dotfiles/vim/bundle/Coqtail/indent/coq.vim
 35: ~/.dotfiles/vim/bundle/Coqtail/syntax/coq.vim
```

**Comment by Iron-E** [comment]
Looking at `:scriptnames` made me realize something: using `vim-plug` made `coqtail` be sourced, but `coqtail` was never sourced if I used `packer` (my plugin manager of choice for 0.5). Following this discovery, I switched from using `packer` to `vim-plug` and it started working right away. After noticing this, I looked at my config for `Coqtail` in both `vim-plug` and `packer`:

```vim
" Vim-plug; this works
Plug 'whonore/Coqtail', {'for': 'coq'}
```

```lua
-- packer; doesn't work
use {'whonore/Coqtail', ft='coq'}
```

So I think this is an issue with `packer` not this plugin. Specifically wbthomason/packer.nvim#93

As for #87, I'll open separate issues for those things to make it easier to track (since they definitely aren't in the scope of this issue!).

I'll close this, since I don't think there's any modifications that can be made here that would alter the functionality of the package manager.
